CODE_STYLE.md

```diff
@@ -0,0 +1,3293 @@
+    body
+  end
+  ```
+
+  One exception to the rule are empty-body methods.
+
+  ```Ruby
+  # good
+  def no_op; end
+  ```
+
+  * Use spaces around operators, after commas, colons and semicolons, around `{` and before `}`. Whitespace might be (mostly) irrelevant to the Ruby interpreter, but its proper use is the key to writing easily readable code.
+
+  ```Ruby
+  sum = 1 + 2
+  a, b = 1, 2
+  [1, 2, 3].each { |e| puts e }
+  class FooError < StandardError; end
+  ```
+
+  The only exception, regarding operators, is the exponent operator:
+
+  ```Ruby
+  # bad
+  e = M * c ** 2
+
+  # good
+  e = M * c**2
+  ```
+
+  `{` and `}` deserve a bit of clarification, since they are used for block and hash literals, as well as string interpolation. For hash literals two styles are considered acceptable.
+
+  ```Ruby
+  # good - space after { and before }
+  { one: 1, two: 2 }
+
+  # good - no space after { and before }
+  {one: 1, two: 2}
+  ```
+
+  The first variant is slightly more readable (and arguably more popular in the Ruby community in general). The second variant has the advantage of adding visual difference between block and hash literals. Whichever one you pick - apply it consistently.
+
+  * No spaces after `(`, `[` or before `]`, `)`.
+
+  ```Ruby
+  some(arg).other
+  [1, 2, 3].size
+  ```
+
+ *  No space after `!`.
+
+  ```Ruby
+  # bad
+  ! something
+
+  # good
+  !something
+  ```
+
+  * No space inside range literals.
+
+    ```Ruby
+    # bad
+    1 .. 3
+    'a' ... 'z'
+
+    # good
+    1..3
+    'a'...'z'
+    ```
+
+* Indent `when` as deep as `case`. I know that many would disagree with this one, but it's the style established in both "The Ruby Programming Language" and "Programming Ruby".
+
+  ```Ruby
+  # bad
+  case
+    when song.name == 'Misty'
+      puts 'Not again!'
+    when song.duration > 120
+      puts 'Too long!'
+    when Time.now.hour > 21
+      puts "It's too late"
+    else
+      song.play
+  end
+
+  # good
+  case
+  when song.name == 'Misty'
+    puts 'Not again!'
+  when song.duration > 120
+    puts 'Too long!'
+  when Time.now.hour > 21
+    puts "It's too late"
+  else
+    song.play
+  end
+  ```
+
+* When assigning the result of a conditional expression to a variable, preserve the usual alignment of its branches.
+
+  ```Ruby
+  # bad - pretty convoluted
+  kind = case year
+  when 1850..1889 then 'Blues'
+  when 1890..1909 then 'Ragtime'
+  when 1910..1929 then 'New Orleans Jazz'
+  when 1930..1939 then 'Swing'
+  when 1940..1950 then 'Bebop'
+  else 'Jazz'
+  end
+
+  result = if some_cond
+    calc_something
+  else
+    calc_something_else
+  end
+
+  # good - it's apparent what's going on
+  kind = case year
+         when 1850..1889 then 'Blues'
+         when 1890..1909 then 'Ragtime'
+         when 1910..1929 then 'New Orleans Jazz'
+         when 1930..1939 then 'Swing'
+         when 1940..1950 then 'Bebop'
+         else 'Jazz'
+         end
+
+  result = if some_cond
```

Comment:
same here, so much whitespace. I'm open to being convinced that this is good, but like the wildly-indented-whens above, it gives me some kind of visual dissonance that feels weird when reading and makes it harder to structure if we _also_ limit ourselves to N-character-wide line length.